bin/sozo/src/commands/calldata_decoder.rs

```diff
@@ -93,6 +94,29 @@ impl CalldataDecoder for ShortStrCalldataDecoder {
     }
 }
 
+/// Decodes a signed integer into a [`Felt`]
+struct SignedIntegerCalldataDecoder;
+impl CalldataDecoder for SignedIntegerCalldataDecoder {
+    fn decode(&self, input: &str) -> DecoderResult<Vec<Felt>> {
+        match input.parse::<i128>() {
+            Ok(value) if value <= i8::MAX as i128 && value >= i8::MIN as i128 => {
+                Ok(vec![Felt::from_i8(value as i8).expect("Invalid numeric string")])
+            }
+            Ok(value) if value <= i16::MAX as i128 && value >= i16::MIN as i128 => {
+                Ok(vec![Felt::from_i16(value as i16).expect("Invalid numeric string")])
+            }
+            Ok(value) if value <= i32::MAX as i128 && value >= i32::MIN as i128 => {
+                Ok(vec![Felt::from_i32(value as i32).expect("Invalid numeric string")])
+            }
+            Ok(value) if value <= i64::MAX as i128 && value >= i64::MIN as i128 => {
+                Ok(vec![Felt::from_i64(value as i64).expect("Invalid numeric string")])
+            }
+            Ok(value) => Ok(vec![Felt::from_i128(value as i128).expect("Invalid numeric string")]),
+            Err(_) => Err(CalldataDecoderError::ParseError("Invalid numeric string".to_string())),
+        }
```

Comment:
Here this can be refactoring to something more simple, something like this should work:
```rust
    fn decode(&self, input: &str) -> DecoderResult<Vec<Felt>> {
        if let Ok(value) = input.parse::<i128>() {
            Ok(vec![value.into()])
        } else {
            Err(CalldataDecoderError::ParseError("Invalid numeric string".to_string()))
        }
    }
```
The reason is that, no matter which signed integer we have, they are all 1 felt long when serialized. 👍